examples/reshare/src/dkg/actor/dealer.rs

```diff
@@ -0,0 +1,206 @@
+/// signed acknowledgements from them, which it gathers into a final log to
+/// put on chain.
+pub struct Actor<E, V, C, S, R>
+where
+    E: Clock + Storage + Metrics,
+    V: Variant,
+    C: Signer,
+{
+    ctx: ContextCell<E>,
+    epoch: Epoch,
+    state: State<E, V, C::PublicKey>,
+    to_players: S,
+    from_players: R,
+    inbox: mpsc::Receiver<Message<V, C>>,
+    dealer: Dealer<V, C>,
+    pub_msg: DealerPubMsg<V>,
+    unsent_priv_msgs: BTreeMap<C::PublicKey, DealerPrivMsg>,
+}
+
+impl<E, V, C, S, R> Actor<E, V, C, S, R>
+where
+    E: Clock + Storage + Metrics + Spawner + CryptoRngCore,
+    V: Variant,
+    C: Signer,
+    S: Sender<PublicKey = C::PublicKey>,
+    R: Receiver<PublicKey = C::PublicKey>,
+{
+    /// Create an [Actor] and its [Mailbox].
+    ///
+    /// `ctx` lets us spawn the actor, and provides randomness.
+    /// `to_players` lets us send messages to the players.
+    /// `from_players` lets us receive messages from the players.
+    /// `round_info` is the configuration for the round.
+    /// `me` is the private key identifying the dealer.
+    /// `share` is the previous share for the dealer.
+    pub async fn init(
+        ctx: E,
+        rng_seed: Summary,
+        state: State<E, V, C::PublicKey>,
+        (to_players, from_players): (S, R),
+        round_info: Info<V, C::PublicKey>,
+        me: C,
+        share: Option<Share>,
+    ) -> (Self, Mailbox<V, C>) {
+        let (outbox, inbox) = mpsc::channel(1);
+        let mailbox = Mailbox(outbox);
+
+        let epoch = Epoch::new(round_info.round());
+        let replay = state.player_acks(epoch).await;
+
+        let (dealer, pub_msg, priv_msgs) = Dealer::start(
+            Transcript::resume(rng_seed).noise(b"dealer-rng"),
+            round_info,
+            me,
+            share,
+        )
+        .expect("should be able to create dealer");
+
+        let mut this = Self {
+            ctx: ContextCell::new(ctx),
+            epoch,
+            state,
+            to_players,
+            from_players,
+            inbox,
+            dealer,
+            pub_msg,
+            unsent_priv_msgs: priv_msgs.into_iter().collect(),
+        };
+
+        for (player, ack) in replay {
+            this.ack(true, player.clone(), ack.clone()).await;
+        }
+
+        (this, mailbox)
+    }
+
+    pub fn start(mut self) -> Handle<()> {
+        spawn_cell!(self.ctx, self.run().await)
+    }
+
+    async fn run(mut self) {
+        let mut finalize = None;
+        select_loop! {
+            self.ctx,
+            on_stopped => {
+                break;
+            },
+            msg = self.from_players.recv() => {
+                let Ok((player, mut msg_bytes)) = msg else {
+                    // The network is dead, so terminate.
+                    break;
+                };
+                let Ok(ack) = PlayerAck::<C::PublicKey>::read(&mut msg_bytes) else {
+                    continue;
+                };
```

Comment:
We should block the player if they send us a malformed acknowledgement

Reply:
What's the thinking behind rate limiting it here, versus just relying on the rate limiting inherited from validator?